src/mame/nec/pc9801.cpp

```diff
@@ -2142,8 +2148,18 @@ void pc9801_atapi_devices(device_slot_interface &device)
 
 void pc9801_state::pc9801_keyboard(machine_config &config)
 {
-	PC9801_KBD(config, m_keyb, 53);
-	m_keyb->irq_wr_callback().set(m_pic1, FUNC(pic8259_device::ir1_w));
+	I8251(config, m_sio_kbd, 0);
+	m_sio_kbd->txd_handler().set("keyb", FUNC(pc9801_kbd_device::input_txd));
+	m_sio_kbd->rxrdy_handler().set(m_pic1, FUNC(pic8259_device::ir1_w));
+
+	clock_device &kbd_clock(CLOCK(config, "kbd_clock", 19'200));
+	kbd_clock.signal_handler().set(m_sio_kbd, FUNC(i8251_device::write_rxc));
+	kbd_clock.signal_handler().append(m_sio_kbd, FUNC(i8251_device::write_txc));
+
+	PC9801_KBD(config, m_keyb, 0);
+	m_keyb->rxd_callback().set("sio_kbd", FUNC(i8251_device::write_rxd));
+	m_keyb->rdy_callback().set("sio_kbd", FUNC(i8251_device::write_cts));
+	m_keyb->rty_callback().set("sio_kbd", FUNC(i8251_device::write_dsr));
```

Comment:
![image](https://github.com/user-attachments/assets/c7bf4582-a277-4d3b-9565-f18e92266e56)

I may have swapped these two, would be very simple to check with schematics I guess ...